chain/store/splitstore/liveset_bloom.go

```diff
@@ -0,0 +1,77 @@
+package splitstore
+
+import (
+	"math/rand"
+
+	"golang.org/x/xerrors"
+
+	bbloom "github.com/ipfs/bbloom"
+	cid "github.com/ipfs/go-cid"
+	blake2b "github.com/minio/blake2b-simd"
+)
+
+const (
+	BloomFilterMinSize     = 10_000_000
+	BloomFilterProbability = 0.01
+)
+
+type BloomLiveSetEnv struct{}
+
+var _ LiveSetEnv = (*BloomLiveSetEnv)(nil)
+
+type BloomLiveSet struct {
+	salt []byte
+	bf   *bbloom.Bloom
+}
+
+var _ LiveSet = (*BloomLiveSet)(nil)
+
+func NewBloomLiveSetEnv() (*BloomLiveSetEnv, error) {
+	return &BloomLiveSetEnv{}, nil
+}
+
+func (e *BloomLiveSetEnv) NewLiveSet(name string, sizeHint int64) (LiveSet, error) {
+	size := int64(BloomFilterMinSize)
+	for size < sizeHint {
+		size += BloomFilterMinSize
+	}
+
+	salt := make([]byte, 4)
+	_, err := rand.Read(salt) //nolint
```

Comment:
We need to use `crypto/rand` here, or seed `math/rand`. Otherwise random values will repeat upon restart.

Reply:
that's totally not a problem :)

Reply:
but did it anyway.